webpack/assets/javascripts/react_app/components/users/PersonalAccessTokens/PersonalAccessTokenModal.js

```diff
@@ -0,0 +1,289 @@
+import React, { useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import {
+  DatePicker,
+  Modal,
+  Radio,
+  ModalVariant,
+  Button,
+  Form,
+  FormGroup,
+  InputGroup,
+  TimePicker,
+  TextInput,
+} from '@patternfly/react-core';
+import PropTypes from 'prop-types';
+import { translate as __ } from '../../../common/I18n';
+import {
+  selectTokens,
+  selectIsSubmitting,
+} from './PersonalAccessTokensSelectors';
+import { APIActions } from '../../../redux/API';
+import { PERSONAL_ACCESS_TOKEN_FORM_SUBMITTED } from './PersonalAccessTokensConstants';
+import './personalAccessToken.scss';
+
+const PersonalAccessTokenModal = ({ controller, url }) => {
+  const dispatch = useDispatch();
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [endsNever, setEndsNever] = useState(true);
+  const [isDateTimeDisabled, setIsDateTimeDisabled] = useState(true);
+  const [name, setName] = useState('');
+  const [date, setDate] = useState('');
+  const [isDateValid, setIsDateValid] = useState(true);
+  const [time, setTime] = useState('');
+  const [isTimeValid, setIsTimeValid] = useState(true);
+  const [showNameErrors, setShowNameErrors] = useState(false);
+
+  const isSubmitting = useSelector(selectIsSubmitting);
+
+  const clearDateTimeState = () => {
+    setDate('');
+    setTime('');
+    setIsDateValid(true);
+    setIsTimeValid(true);
+  };
+
+  const handleModalToggle = () => {
+    setIsModalOpen(!isModalOpen);
+    setName('');
+    setShowNameErrors(false);
+    clearDateTimeState();
+    setIsDateTimeDisabled(true);
+    setEndsNever(true);
+  };
+
+  const validateDateChange = newDate => {
+    if (
+      !newDate.length ||
+      (newDate.split('-').length === 3 &&
+        newDate &&
+        !Number.isNaN(new Date(newDate).getTime()))
+    ) {
+      setIsDateValid(true);
+    } else {
+      setIsDateValid(false);
+    }
+    setDate(newDate);
+  };
+
+  const validateTimeChange = newTime => {
```

Comment:
not sure why, but when I enter time with only `hh:mm` there is a error in the time, but without text
![Screenshot from 2023-05-04 10-52-02](https://user-images.githubusercontent.com/30431079/236156922-e0b1865c-7b1f-4aa8-8ab8-485f45b27aa1.png)